pkg/ingester/ingester.go

```diff
@@ -809,17 +809,15 @@ func (i *Ingester) PushWithCleanup(ctx context.Context, pushReq *push.Request) (
 
 	db, err := i.getOrCreateTSDB(userID, false)
 	if err != nil {
-		// Check for a particular per-instance limit and return that error directly
-		// since it contains extra information for gRPC and our logging middleware.
-		if errors.Is(err, errMaxTenantsReached) {
-			return nil, err
-		}
-		return nil, annotateWithUser(err, userID)
+		// If this is a safe error, we wrap it with userID and return it, because
+		// it might contain extra information for gRPC and our logging middleware.
```

Comment:
I would suggest to move also this comment to the function, in order to not repeat it at every call.